### PR TITLE
DL-2054 Auth upgrade for ACM-FE

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-2054 Auth upgrade for ACM-FE

**Auth upgrade**

* Upgraded ACM-FE to the latest `auth-client` library.

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientmandate/connectors/AgentClientMandateConnector.scala [hunk 1]
```diff
@@ -47,117 +46,120 @@ trait AgentClientMandateConnector extends ServicesConfig with RawResponseReads {
...
     http.POST[JsValue, HttpResponse](postUrl, jsonData)
   }
 
-  def fetchAllMandates(arn: String, serviceName: String, allClients: Boolean, displayName: Option[String])(implicit hc: HeaderCarrier, ac: AuthContext): Future[HttpResponse] = {
-    val authLink = AuthUtils.getAuthLink
+  def fetchAllMandates(agentAuthRetrievals: AgentAuthRetrievals, serviceName: String, allClients: Boolean, displayName: Option[String])
+                      (implicit hc: HeaderCarrier): Future[HttpResponse] = {
+
+    val AgentAuthRetrievals(arn, agentCode, _, _, _) = agentAuthRetrievals
+    val authLink = s"/agent/$agentCode"
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/agentclientmandate/connectors/AgentClientMandateConnector.scala]: AMBER - Not sure why this is needed? You have the mandateConnctorUri already made.

Comment (Ufuoma-Daniel) [reply app/uk/gov/hmrc/agentclientmandate/connectors/AgentClientMandateConnector.scala]: applies to all other instances of this too ^

File: app/uk/gov/hmrc/agentclientmandate/controllers/agent/AgencyDetailsController.scala [hunk 2]
```diff
@@ -16,43 +16,40 @@
...
-
-
 }
 
 object AgencyDetailsController extends AgencyDetailsController {
-  // $COVERAGE-OFF$
+
   val dataCacheService: DataCacheService = DataCacheService
-  val authConnector: AuthConnector = FrontendAuthConnector
+  val authConnector: AuthConnector = ConcreteAuthConnector
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/agentclientmandate/controllers/agent/AgencyDetailsController.scala]: GREY -ConcreteAuthConnector? not sure I'm a fan of this name.

File: app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      Logger.warn(s"[authErrorHandling] Call to auth failed - $e")
+      InternalServerError
+    case e: AuthorisationException =>
+      Logger.info(s"[authErrorHandling] Authorisation exception of type: - ${e.getClass.toString}")
+      Redirect(loginUrl, loginParams(service.getOrElse(""), isAnAgent))
+  }
+
+  def agentAuthenticated[A](service: Option[String], retrieval: Retrieval[A])(body: A => Future[Result])
+                           (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
+    authorised(Enrolment(agentRefEnrolment) and AuthProviders(GovernmentGateway)).retrieve(retrieval) {
```
Comment (Ufuoma-Daniel) [inline app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala]: RED? - I'm not sure about checking for the Agent enrolment is the same as checking for the Agent affinity group.

File: app/uk/gov/hmrc/agentclientmandate/controllers/agent/AgencyDetailsController.scala (hunk 2 above)
Comment (paulanderson1234) [reply app/uk/gov/hmrc/agentclientmandate/controllers/agent/AgencyDetailsController.scala]: Also found in PerformanceTestSupportController and ApplicationController where we are using FrontendAuthConnector as opposed to ConcreteAuthConnector

File: app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala (hunk 3 above)
Comment (jmountain-hmrc) [reply app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala]: Well, I don't believe a non-agent would have the Agent enrolment - if it's more reasssuring I can include a check for the affinity group

File: app/uk/gov/hmrc/agentclientmandate/connectors/AgentClientMandateConnector.scala (hunk 1 above)
Comment (jmountain-hmrc) [reply app/uk/gov/hmrc/agentclientmandate/connectors/AgentClientMandateConnector.scala]: It depends if it's in scope for the function. Sometimes the routes are agent only so there's no point calling the function

File: app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala (hunk 3 above)
Comment (Ufuoma-Daniel) [reply app/uk/gov/hmrc/agentclientmandate/controllers/auth/AuthorisedWrappers.scala]: Yes please, at least for the consistency with the other services.

**Pull request merged**